rclrs/src/lib.rs

```diff
@@ -81,45 +81,37 @@ pub fn spin(node: &node::Node) -> Result<(), WaitSetErrorResponse> {
 ///         +--------------------+
 ///
 ///
-pub fn spin_once(node: &Node, timeout_ns: i64) -> Result<(), WaitSetErrorResponse> {
+pub fn spin_once(node: &Node, timeout_ns: i64) -> Result<(), RclReturnCode> {
     let number_of_subscriptions = node.subscriptions.len();
     let number_of_guard_conditions = 0;
     let number_of_timers = 0;
     let number_of_clients = 0;
     let number_of_services = 0;
     let number_of_events = 0;
 
-    let context = &mut *node.context.lock();
-
     let mut wait_set = WaitSet::new(
         number_of_subscriptions,
         number_of_guard_conditions,
         number_of_timers,
         number_of_clients,
         number_of_services,
         number_of_events,
-        context,
+        &mut *node.context.lock(),
     )?;
 
-    for subscription in &node.subscriptions {
-        match wait_set.add_subscription(subscription) {
-            Ok(()) => (),
-            Err(WaitSetErrorResponse::DroppedSubscription) => (),
-            Err(err) => return Err(err),
-        };
+    let live_subscriptions = node.live_subscriptions();
+    for sub in &live_subscriptions {
+        wait_set.add_subscription(&**sub)?;
     }
 
     wait_set.wait(timeout_ns)?;
-    for (i, sub) in node.subscriptions.iter().enumerate() {
+    for (i, sub) in live_subscriptions.iter().enumerate() {
```

Comment:
Please avoid shortened names, rename `sub` to `live_subscription`

Reply:
Done (though I personally don't mind the shorter names as it is visually more distinct from the `live_subscriptions` variable).